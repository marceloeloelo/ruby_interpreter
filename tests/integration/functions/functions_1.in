def method_1
end

def method_2 x
end

def method_3 x, y
end

def method_4(x)
end

def method_5(x,y,z)
end

