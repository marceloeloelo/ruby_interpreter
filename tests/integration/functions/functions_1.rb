def method_1
end

def method_2 x
  x
end

def method_3 x, y
  y
end

def method_4(x)
  "hola"
end

def method_5(x,y,z)
  x
  y
  z
end

x = y = z = 5

method_1()

method_2(x)
puts(method_2(x))
puts(method_2(1))

method_3(x, y)
puts(method_3(x, y))
puts(method_3(2, y))

method_4(x)
puts(method_4(x))
puts(method_4(2))

method_5(x, y, z)
puts(method_5(x, y, z))
puts(method_5(2, y, 1))
