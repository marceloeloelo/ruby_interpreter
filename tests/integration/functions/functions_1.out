5
1
5
5
hola
hola
5
1
